vuu-ui/packages/vuu-shell/src/layout-management/useLayoutManager.tsx

```diff
@@ -1,42 +1,47 @@
 import React, { useState, useCallback, useContext, useEffect } from "react";
-import { getLocalEntity, saveLocalEntity } from "@finos/vuu-filters";
-import { LayoutJSON } from "@finos/vuu-layout";
-import { getUniqueId } from "@finos/vuu-utils";
+import { getLocalEntity } from "@finos/vuu-filters";
+import { LayoutJSON, LayoutPersistenceManager } from "@finos/vuu-layout";
 import { LayoutMetadata, Layout } from "./layoutTypes";
 
 export const LayoutManagementContext = React.createContext<{
   layouts: Layout[],
-  saveLayout: (n: Omit<LayoutMetadata, "id">) => void
+  saveLayout: (n: LayoutMetadata) => void
 }>({ layouts: [], saveLayout: () => { } })
 
-export const LayoutManagementProvider = (props: { children: JSX.Element | JSX.Element[] }) => {
-
-  const [layouts, setLayouts] = useState<Layout[]>([])
+export const LayoutManagementProvider = (props: {
+    persistenceManager: LayoutPersistenceManager,
+    children: JSX.Element | JSX.Element[]
+  }) => {
+  const [layouts, setLayouts] = useState<Layout[]>([]);
+  const [layout, setLayout] = useState<LayoutJSON>();
+  const [metadata, setMetadata] = useState<LayoutMetadata>();
 
   useEffect(() => {
-    const layouts = getLocalEntity<Layout[]>("layouts")
-    setLayouts(layouts || [])
+    const loadedLayouts = props.persistenceManager.loadLayouts();
+    setLayouts(loadedLayouts || [])
   }, [])
 
   useEffect(() => {
```

Comment:
do we need this useEffect here? could this code just be appended to saveLayout?

Reply:
discussed offline: we probably dont need this useEffect and state for layout and metadata